client/my-sites/site-settings/publishing-tools/post-by-email.tsx

```diff
@@ -0,0 +1,160 @@
+
+	const handleRegenerate = () => {
+		if ( ! selectedSiteId ) {
+			return;
+		}
+
+		regeneratePostByEmail( selectedSiteId );
+	};
+
+	const email = address && address !== 'regenerate' ? address : '';
+	const labelClassName =
+		moduleUnavailable || regeneratingPostByEmail || ! active ? 'is-disabled' : undefined;
+
+	return (
+		<>
+			<FormFieldset>
+				<SupportInfo
+					text={ translate(
+						'Allows you to publish new posts by sending an email to a special address.'
+					) }
+					link={
+						siteIsAtomic
+							? localizeUrl( 'https://wordpress.com/support/post-by-email/' )
+							: 'https://jetpack.com/support/post-by-email/'
+					}
+					privacyLink={ ! siteIsAtomic }
+				/>
+				{ siteIsJetpack ? (
+					<JetpackModuleToggle
+						siteId={ selectedSiteId }
+						moduleSlug="post-by-email"
+						label={ translate( 'Publish posts by sending an email' ) }
+						disabled={ isFormPending || moduleUnavailable }
+					/>
+				) : (
+					<ToggleControl
+						checked={ active }
+						disabled={ isFormPending }
+						label={ translate( 'Post by Email' ) }
+						onChange={ ( checked ) => {
+							if ( ! selectedSiteId ) {
+								return;
+							}
+
+							if ( checked ) {
+								activateModule( selectedSiteId, moduleSlug );
+							} else {
+								deactivateModule( selectedSiteId, moduleSlug );
+							}
+						} }
+					/>
+				) }
+
+				<div className="publishing-tools__module-settings site-settings__child-settings">
+					<FormLabel className={ labelClassName }>
+						{ translate( 'Send your new posts to this email address:' ) }
+					</FormLabel>
+					<ClipboardButtonInput
+						className="publishing-tools__email-address"
+						disabled={ regeneratingPostByEmail || ! active || moduleUnavailable }
+						value={ email }
+					/>
+					<Button
+						onClick={ handleRegenerate }
+						disabled={
+							isFormPending || regeneratingPostByEmail || ! active || !! moduleUnavailable
+						}
+					>
+						{ regeneratingPostByEmail
+							? translate( 'Regenerating…' )
+							: translate( 'Regenerate address' ) }
+					</Button>
+				</div>
+			</FormFieldset>
+		</>
+	);
+};
+
+export const PostByEmailSetting = connect(
+	( state: IAppState ) => {
+		const selectedSiteId = getSelectedSiteId( state ) || 0;
+		const siteIsJetpack = isJetpackSite( state, selectedSiteId );
+		const isAtomic = isSiteAutomatedTransfer( state, selectedSiteId );
+		const regeneratingPostByEmail = isRegeneratingJetpackPostByEmail( state, selectedSiteId );
+		const active = !! isJetpackModuleActive( state, selectedSiteId, moduleSlug );
+		const siteInDevMode = isJetpackSiteInDevelopmentMode( state, selectedSiteId );
+		const moduleUnavailableInDevMode = isJetpackModuleUnavailableInDevelopmentMode(
+			state,
+			selectedSiteId,
+			moduleSlug
+		);
+
+		return {
+			selectedSiteId,
+			siteIsJetpack,
+			isAtomic,
+			regeneratingPostByEmail,
+			active,
+			moduleUnavailable: siteInDevMode && moduleUnavailableInDevMode,
+		};
```

Comment:
Same as here https://github.com/Automattic/wp-calypso/pull/90489#discussion_r1603539659, we should use `useSelector` from `calypso/state` instead of `connect`s `mapStateToProps`.

Reply:
@fredrikekelund In my case it was just 1 property and I thought that it was the reason to get rid of `connect`. But looks like the reason is something else.
Is there a recommendation from A8C to avoid using `connect` or maybe some other reason? Since, personally, for me, `connect` is quite good tool.

And with the help of `useSelector` it does not look terse when we copy variable names 3 times, doing simple things.
https://github.com/Automattic/wp-calypso/pull/90819/files#diff-1a2d2a711ac2389148e7764b72177d565e3e6a88fea709f208906a031a5a64c5R36-R64


@chad1008 Let's at least avoid redundant variable renamings:
```
const {
		selectedSiteId,
        ...
	} = useSelector( ( state ) => {
		const selectedSiteId = getSelectedSiteId( state ) || 0;
        ....

		return {
			selectedSiteId,
			....
		};
	} );
```

Instead of:
```
const {
		siteId: selectedSiteId,
        ...
	} = useSelector( ( state ) => {
		const siteId = getSelectedSiteId( state ) || 0;
        ....

		return {
			siteId,
			....
		};
	} );
```

Reply:
Oh yes. I have a lot of variable renaming to do. Stuff like this, but I'm also going through and making sure it's clear which flow (Simple vs Jetpack) individual variables are a part of.

Reply:
I left a comment about this here https://github.com/Automattic/wp-calypso/pull/90819#discussion_r1611276938

Reply:
> Is there a recommendation from A8C to avoid using connect or maybe some other reason? Since, personally, for me, connect is quite good tool.

`connect` isn't deprecated, it's just that hooks are recommended as the default (see [docs](https://react-redux.js.org/api/connect)).

This post (even if it's old) has some Calypso-specific background p4TIVU-9k1-p2
In short, `connect` is a viable alternative for more complex cases. For this particular PR, I think we should stick with `useSelector`.